tests/cpp/test_alias.cpp

```diff
@@ -1395,4 +1395,28 @@ TEST_F(AliasTest, SegmentMetaOps) {
   }
 }
 
+TEST_F(AliasTest, Issue2552) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* x = makeContigConcreteTensor({1, 3});
+  TensorView* y = makeContigConcreteTensor({1, 3});
+  fusion->addInput(x);
+  fusion->addInput(y);
+  x = expand(x, {IrBuilder::create<Val>(2), x->axis(1)->extent()});
+  x = slice(x, /*starts=*/{0, 0}, /*stops=*/{1, 3});
```

Comment:
This works for the example. I tried to play with a variation where we expand to something big and slice down to a non-broadcast size, this seems to break for me.

```
  x = expand(x, {IrBuilder::create<Val>(8), x->axis(1)->extent()});
  x = slice(x, /*starts=*/{0, 0}, /*stops=*/{5, 3});
```
Can we also try something like this?

Reply:
Indeed. Sounds like a different bug, in MarkAliasesPrepare: 

```
C++ exception with description "The contiguity of a broadcast/reduction dimension must be None. The contiguity of a non-broadcast/reduction dimension must be true/false
Exception raised from validateContiguity at /opt/pytorch/nvfuser/csrc/ir/nodes.cpp:2960 (most recent call first):
```

Reply:
Feel free to merge this PR as-is and we (or I) can follow up in a different PR.

Reply:
#2555